src/backend/distributed/utils/resource_lock.c

```diff
@@ -1357,7 +1357,8 @@ AcquireDistributedLockOnRelations(List *relationList, LOCKMODE lockMode, uint32
 													 (void *) lockRelationRecord);
 		}
 
-		if ((configs & DIST_LOCK_REFERENCING_TABLES) > 0)
+		if (get_rel_relkind(relationId) == RELKIND_RELATION &&
```

Comment:
Q: should we check for RELKIND_PARTITIONED_TABLE too, and maybe for RELKIND_FOREIGN_TABLE ?

Reply:
I think you can switch to testing for relkind != RELKING_VIEW to cover all the non view cases.